elastic/beats
# Add support for multiple sets of hints on autodiscover

Enhancement

## What does this PR do?

This PR allows defining multiple sets of annotations similar to how processors are defined. This functionality already exists in heartbeat. This PR standardizes the utility and adds it to logs and metrics hints builder as well. 

## Why is it important?

A container can always have more than one kind of endpoint that requires different kinds of processing needs. We need to provide a uniform experience across all beats from a hints perspective. 

## Checklist


## Author's Checklist



## How to test this PR locally

- Check for regressions:
  - Use heartbeat with hints enabled and setup a pod with multiple monitors, this should continue working
  - Use metricbeat or filebeat with hints enabled and setup a pod with hints as usually configured, this should continue working.
  - Repeat the previous test with container-specific hints.
- Test the new feature:
  - Use metricbeat or filebeat with hints enabled, and setup a pod with multiple sets of hints (there are some examples in the docs added in this PR), check that multiple modules are instantiated for the pod.
  - Repeat the previous test with container-specific hints.


## Related issues

- #13119 - PR where this feature was added for Heartbeat

## Use cases
